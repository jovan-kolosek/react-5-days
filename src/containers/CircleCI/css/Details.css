.details-container {
  position: fixed;
  top: 35%;
  left: 50%;
  transform: translate(-50%, -50%);
  -webkit-transform: translate(-50%, -50%);
  -moz-transform: translate(-50%, -50%);
  -o-transform: translate(-50%, -50%);
  -ms-transform: translate(-50%, -50%);
  
  font-size: 20px;
  padding: 5px;
  width: 500px;
}

.details_container__header {
  text-align: center;
}

.details_container__info {
  padding: 8px;
}

.details_container__build {
  padding: 20px;
  padding-left: 30px;
  padding-bottom: 25px;
}

.details_container__commit {
  padding: 8px;
  padding-left: 30px;
}
