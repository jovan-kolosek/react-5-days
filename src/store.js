import { applyMiddleware, createStore, compose } from 'redux';
import rootReducer from './reducers';
import thunk from 'redux-thunk';
import { logger } from './middleware';

const middleware = [thunk, logger];

const store = compose(applyMiddleware(...middleware))(createStore)(rootReducer);

export default store;
