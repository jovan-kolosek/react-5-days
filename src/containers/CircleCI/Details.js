import React, {Component} from 'react';
import { connect } from 'react-redux';

import './css/Details.css'

class Details extends Component {

  render() {
    const { selectedBuild } = this.props
    return (
      <div className="details-container">
        <h1 className="details_container__header"><b>Circle CI</b></h1>
        <div className="details_container__build"><b>Build no #{selectedBuild.build_num}</b></div>
        <div className="details_container__info"><b>Project name: {selectedBuild.reponame}</b></div>
        <div className="details_container__info"><b>Branch name: {selectedBuild.branch}</b></div>
        <div className="details_container__info"><b>Status: {selectedBuild.status}</b></div>
        <div className="details_container__info"><b>Commit message:</b></div>
        <div className="details_container__commit">{selectedBuild.subject}</div>
      </div>
    );
  }
}

function stateToProps(state) {
  return {
    selectedBuild: state.builds.selectedBuild,
  };
}

const Details_Connected = connect(stateToProps)(Details);

export default Details_Connected;
