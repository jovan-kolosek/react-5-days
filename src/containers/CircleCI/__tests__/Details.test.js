import React from 'react';
import Details from '../Details';
import { Provider } from 'react-redux';
import store from '../../../store';
import renderer from 'react-test-renderer';

test('Details with default context renders', () => {
  const component = renderer.create(
    <Provider store={store}><Details match={{params: {}}} /></Provider>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});