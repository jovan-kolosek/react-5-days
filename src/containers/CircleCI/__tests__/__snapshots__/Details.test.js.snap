// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details with default context renders 1`] = `
<div
  className="details-container"
>
  <div>
    Loading data...
  </div>
</div>
`;
